4 - AnimalShelter/Code/Shelter.cs

```diff
@@ -0,0 +1,106 @@
+﻿using AnimalShelter.Code.Enums;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace AnimalShelter.Code
+{
+    public class Shelter
+    {
+        public Dictionary<Guid?, Animal> StoredAnimals = new Dictionary<Guid?, Animal>();
```

Comment:
Good use of using Guids but allowing your Key of your dictionary to be null-able is *dangerous* and something you should never do. Your key should never be something you want null. Guid's in general, is never something you want to be null.

Reply:
To add to this, to me the unique key should be apart of the animal and not something stored else where. Normally your "id" is apart of the object.